src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+    }
+  }
+  else if (_num_polar == 12) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.1252334085);
+      thetas[a*(_num_polar/2)+1] = acos(0.3678314989);
+      thetas[a*(_num_polar/2)+2] = acos(0.5873179542);
+      thetas[a*(_num_polar/2)+3] = acos(0.7699026741);
+      thetas[a*(_num_polar/2)+4] = acos(0.9041172563);
+      thetas[a*(_num_polar/2)+5] = acos(0.9815606342);
+    }
+  }
+
+  /* Set the arrays of sin thetas and weights */
+  Quadrature::setThetas(thetas, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] thetas;
+}
+
+
+void GLPolarQuad::precomputeWeights(bool solve_3D) {
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  FP_PRECISION* weights = new FP_PRECISION[_num_polar/2*_num_azim/4];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Leonard polar angle quadrature */
+  if (_num_polar == 2) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 1.0;
+    }
+  }
+  else if (_num_polar == 4) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.6521451549 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.3478548451 / 2.0;
+    }
+  }
+  else if (_num_polar == 6) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.4679139346 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.3607615730 / 2.0;
+      weights[a*(_num_polar/2)+2] = 0.1713244924 / 2.0;
+    }
+  }
+  else if (_num_polar == 8) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.3626837834 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.3137066459 / 2.0;
+      weights[a*(_num_polar/2)+2] = 0.2223810344 / 2.0;
+      weights[a*(_num_polar/2)+3] = 0.1012285363 / 2.0;
+    }
+  }
+  else if (_num_polar == 10) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.2955242247 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.2692667193 / 2.0;
+      weights[a*(_num_polar/2)+2] = 0.2190863625 / 2.0;
+      weights[a*(_num_polar/2)+3] = 0.1494513492 / 2.0;
+      weights[a*(_num_polar/2)+4] = 0.0666713443 / 2.0;
+    }
+  }
+  else if (_num_polar == 12) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.2491470458 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.2334925365 / 2.0;
+      weights[a*(_num_polar/2)+2] = 0.2031674267 / 2.0;
+      weights[a*(_num_polar/2)+3] = 0.1600783286 / 2.0;
+      weights[a*(_num_polar/2)+4] = 0.1069393260 / 2.0;
+      weights[a*(_num_polar/2)+5] = 0.0471753364 / 2.0;
+    }
+  }
+
+  /* Set the arrays of sin thetas and weights */
+  Quadrature::setPolarWeights(weights, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] weights;
+
+  Quadrature::precomputeWeights(solve_3D);
+}
+
+
+
+/**
+ * @brief Dummy constructor calls the parent constructor.
+ */
+EqualWeightPolarQuad::EqualWeightPolarQuad(): Quadrature() {
+  _quad_type = EQUAL_WEIGHT;
```

Comment:
ditto, perhaps use a default number of polar angles here so that `initialize()` won't errantly complete?